src/bin/pg_autoctl/system_utils.c

```diff
@@ -73,26 +74,33 @@ get_system_info_linux(SystemInfo *sysInfo)
 
 
 /*
- * FreeBSD, OpenBSD, and darwin use the sysctlbyname(3) API.
+ * FreeBSD, OpenBSD, and darwin use the sysctl(3) API.
  */
 #if defined(__APPLE__) || defined(BSD)
 static bool
 get_system_info_bsd(SystemInfo *sysInfo)
 {
 	unsigned int ncpu = 0;      /* the API requires an integer here */
+	int mib[2];
 
 	size_t cpuSize = sizeof(ncpu);
 	size_t memSize = sizeof(sysInfo->totalram);
 
-	if (sysctlbyname("hw.ncpu", &ncpu, &cpuSize, NULL, 0) != 0)
+	mib[0] = CTL_HW;
+	mib[1] = HW_NCPU;
+
+	if (sysctl(mib, 2, &ncpu, &cpuSize, NULL, 0) == -1)
 	{
 		log_error("Failed to probe number of CPUs: %m");
 		return false;
 	}
 
 	sysInfo->ncpu = (unsigned short) ncpu;
 
-	if (sysctlbyname("hw.memsize", &(sysInfo->totalram), &memSize, NULL, 0) != 0)
+	mib[0] = CTL_HW;
+	mib[1] = HW_PHYSMEM;
+
```

Comment:
What about using the following style, that would look more like what we use in the rest of the codebase?

``` c
    int ramMIB[2] = { CTL_HW, HW_PHYSMEM };
```

Reply:
Good suggestion; will do